Hide the download link for ICS files when printing

matthillco opened the pull request:
## What

Hide the download link for ICS files when printing any page containing this component.

Example page affected: https://www.gov.uk/bank-holidays

[Trello card](https://trello.com/c/PnEVIZtq/192-review-and-fix-application-component-print-styles), [Jira issue PNP-8589](https://gov-uk.atlassian.net/browse/PNP-8589)

## Why
Download links for ICS files are redundant on a printed page, since they are interactive elements and make no sense to be printed. Hiding them when printing improves the clarity of the printed output.

## Visual Changes
### BEFORE
<img width="1180" alt="image" src="https://github.com/user-attachments/assets/6bbd3066-a97d-4d60-bedf-e7e92edbfcc4">

### AFTER
<img width="1180" alt="image" src="https://github.com/user-attachments/assets/4ffd12f6-2392-47b1-ba31-3b3335767ce5">

matthillco committed:
Hide the `subscribe` component when printing

andysellick reviewed (commented):
Please can you change the PR description to explain what specifically this change is, why it's being made and on what page(s) it has an impact.

matthillco commented:
> Please can you change the PR description to explain what specifically this change is, why it's being made and on what page(s) it has an impact.

I've updated the PR title, description, and added before/after images.
